### PR TITLE
Add string utils file

### PR DESCRIPTION
**Pull request by filippobrizzi**
# Description
Extract string utils into a `string_utils.h` file.

### CODE REVIEW

File: modules/utils/include/hephaestus/utils/string_utils.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//=================================================================================================
+
+#pragma once
+
+#include <string>
+
+namespace heph::utils {
+
+/// Truncates a string by returning the segment between start and end token strings, including
+/// the start token but excluding the end token. For example, the following code will return
```
[the remark this answers is not in the document]
Comment (filippobrizzi) [reply modules/utils/include/hephaestus/utils/string_utils.h]: Add a flag to decide if end should be included or not

File: modules/utils/include/hephaestus/utils/string_utils.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/// 'to/some/file'
+/// @code
+/// constexpr auto str = "/path/to/some/file.txt";
+/// constexpr auto start_token = "to";
+/// constexpr auto end_token = ".txt";
+/// constexpr auto truncated = truncate(str, start_token, end_token);
+/// std::cout << truncated << '\n';
+/// @endcode
+constexpr auto truncate(std::string_view str, std::string_view start_token,
```
[the remark this answers is not in the document]
Comment (filippobrizzi) [reply modules/utils/include/hephaestus/utils/string_utils.h:23]: what do you mean with non-unique tokens?

[the remark this answers is not in the document]
Comment (lhruby) [reply modules/utils/include/hephaestus/utils/string_utils.h:23]: e.g. `path/to/object/or/to/folder.txt`, start: `to`, end `folder`

File: modules/utils/include/hephaestus/utils/string_utils.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhruby) [reply modules/utils/include/hephaestus/utils/string_utils.h]: Hehe, I just meant to extend the description by the interval. The flag is a nice solution though.

File: modules/utils/include/hephaestus/utils/string_utils.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (filippobrizzi) [reply modules/utils/include/hephaestus/utils/string_utils.h:23]: good point! added a comment

**Review by lhruby — APPROVED** [review]

**Pull request merged**